### ISSUE TITLE
Support for PostgreSQL Flexible Server storage auto-grow

### ISSUE THREAD

**Issue by marcindulak** [hashicorp/terraform-provider-azurerm#22699]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Description

Please support PostgreSQL Flexible Server [storage auto-grow](https://learn.microsoft.com/en-us/azure/postgresql/flexible-server/concepts-compute-storage#storage-auto-grow-preview).

### New or Affected Resource(s)/Data Source(s)

azurerm_postgresql_flexible_server

### Potential Terraform Configuration

```hcl
resource "azurerm_postgresql_flexible_server" "example" {
...
storage_auto_grow_enabled = true
...
}
```

### References

https://techcommunity.microsoft.com/t5/azure-database-for-postgresql/storage-autogrow-and-online-disk-scaling-in-azure-postgresql/ba-p/3868491

**Comment by ezeql** [comment]
Property name must probably be 'auto_grow_enabled', matching MySQL 'https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/mysql_flexible_server#auto_grow_enabled

**Comment by SebastianBalle** [comment]
Looks to be implemented in [23069](https://github.com/hashicorp/terraform-provider-azurerm/pull/23069)

**Comment by rcskosir** [comment]
Thanks for taking the time to submit this issue. It looks like this has been resolved as of #23069. As such, I am going to mark this issue as closed. If that is not the case, please provide additional information regarding the enhancement, thanks!
